lucene/core/src/java/org/apache/lucene/codecs/lucene87/DeflateWithPresetDictCompressionMode.java

```diff
@@ -155,16 +155,15 @@ public Decompressor clone() {
 
   private static class DeflateWithPresetDictCompressor extends Compressor {
 
-    final byte[] dictBytes;
-    final int blockLength;
+    private final int dictLength, blockLength;
     final Deflater compressor;
     byte[] compressed;
     boolean closed;
 
     DeflateWithPresetDictCompressor(int level, int dictLength, int blockLength) {
-      compressor = new Deflater(level, true);
```

Comment:
Just a thought, really.

If it's a bug that can be probed for (and it can be - see Adrian's repro) then it could as well be a static initialization of a supplier of Deflater instances; if we probe for a buggy JVM, we return the wrapper. If we don't we return the Deflater. This way on non-affected JVMs nothing happens and if we do use the wrapper, we know the JVM is broken.

Reply:
that's also my idea! I am working on probing at the moment. like 128 bytes of dict and 128 bytes of data. If result is correct the factory method i implemented uses the plain Deflater.

Reply:
I added detection in latest commit. Works with 8 bytes :-)